compass/setup.py

```diff
@@ -436,6 +436,10 @@ def _get_basic_config(config_file, machine, mpas_model_path, mpas_core):
         machine = 'default'
     config.add_from_package('compass.machines', f'{machine}.cfg')
 
+    # store name of machine so tests have access to it if needed
+    if machine is not None:
+        config.set('deploy', 'machine', machine)
+
```

Comment:
I think this is great!  There is also a `$COMPASS_MACHINE` environment variable set in the load script, so that would have been another way that maybe I didn't think of at the time.